examples/calibration/plot_calibration_curve.py

```diff
@@ -5,130 +5,262 @@
+brier = []
+precision = []
+recall = []
+f1 = []
 
-    fig = plt.figure(fig_index, figsize=(10, 10))
-    ax1 = plt.subplot2grid((3, 1), (0, 0), rowspan=2)
-    ax2 = plt.subplot2grid((3, 1), (2, 0))
+for i, (clf, name) in enumerate(clf_list):
+    clf.fit(X_train, y_train)
+    y_pred = clf.predict(X_test)
 
-    ax1.plot([0, 1], [0, 1], "k:", label="Perfectly calibrated")
-    for clf, name in [(lr, 'Logistic'),
-                      (est, name),
-                      (isotonic, name + ' + Isotonic'),
-                      (sigmoid, name + ' + Sigmoid')]:
-        clf.fit(X_train, y_train)
-        y_pred = clf.predict(X_test)
-        if hasattr(clf, "predict_proba"):
-            prob_pos = clf.predict_proba(X_test)[:, 1]
-        else:  # use decision function
-            prob_pos = clf.decision_function(X_test)
-            prob_pos = \
-                (prob_pos - prob_pos.min()) / (prob_pos.max() - prob_pos.min())
+    # Create DataFrame index
+    index[i] = name
+    # Store column data
+    brier.append(brier_score_loss(y_test, y_pred))
+    precision.append(precision_score(y_test, y_pred))
+    recall.append(recall_score(y_test, y_pred))
+    f1.append(f1_score(y_test, y_pred))
 
-        clf_score = brier_score_loss(y_test, prob_pos, pos_label=y.max())
-        print("%s:" % name)
-        print("\tBrier: %1.3f" % (clf_score))
-        print("\tPrecision: %1.3f" % precision_score(y_test, y_pred))
-        print("\tRecall: %1.3f" % recall_score(y_test, y_pred))
-        print("\tF1: %1.3f\n" % f1_score(y_test, y_pred))
+score_df = pd.DataFrame(
+    data={'Brier score': brier, 'Precision': precision, 'Recall': recall,
+          'F1': f1},
+    index=index,
+)
+score_df
 
-        fraction_of_positives, mean_predicted_value = \
-            calibration_curve(y_test, prob_pos, n_bins=10)
+# %%
+# Next, we will compare:
+#
+# * :class:`~sklearn.linear_model.LogisticRegression` (baseline)
+# * :class:`~sklearn.svm.LinearSVC` (:term:`decision_function` scores
+#   min-max scaled to [0,1], since they are not proper probabilities)
+# * :class:`~sklearn.svm.LinearSVC` with isotonic and sigmoid
+#   calibration (see :ref:`User Guide <calibration>`)
+#
 
-        ax1.plot(mean_predicted_value, fraction_of_positives, "s-",
-                 label="%s (%1.3f)" % (name, clf_score))
 
-        ax2.hist(prob_pos, range=(0, 1), bins=10, label=name,
-                 histtype="step", lw=2)
+class NaivelyCalibratedLinearSVC(LinearSVC):
+    """LinearSVC with `predict_proba` method that naively scales
+    `decision_function` output."""
 
-    ax1.set_ylabel("Fraction of positives")
-    ax1.set_ylim([-0.05, 1.05])
-    ax1.legend(loc="lower right")
-    ax1.set_title('Calibration plots  (reliability curve)')
+    def predict_proba(self, X):
+        """min-max scale output of `decision_function` to [0,1]."""
+        df = self.decision_function(X)
+        calibrated_df = (df - df.min()) / (df.max() - df.min())
+        return calibrated_df
```

Comment:
In retrospect, this way of calibrating is overly naive, in the sense that if we predict one sample at time it is undefined (0/0).

Maybe we should record `self,.df_min_` and `self.df_max_` in an overloaded fit method on the training set and use that to


```python
    def fit(self, X, y):
        super().fit(X, y)
        df = self.decision_function(X)
        self.df_min_ = df.min()
        self.df_max_ = df.max()

    def predict_proba(self, X):
        """min-max scale output of `decision_function` to [0, 1]."""
        df = self.decision_function(X)
        calibrated_df = (df - self.df_min_) / (self.df_max_ - self.df_min_)
        return np.clip(calibrated_df, 0, 1)
```